Bottom: 13c148f833b9e2e711e487c36233fc853cbe5b98
Top:    db84443cd4eeeb112ee7a02e58f49fb0f5c4cead
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 21:11:39 +0000

Tidy up range check in misc.c

For consistnecy with other range checks involving max_copper_layer, and to
drive home the fact we are checking for exceeding max_copper_layer + 2.
NB: The +2 relates to the bottom silk and top-silk layers, whose numbers
are used to denote the bottom and top layer groups respectively.

re-write:
  layer > max_copper_layer + 1

to become:
  layer >= max_copper_layer + 2


---

diff --git a/src/misc.c b/src/misc.c
index 4f381ae..37aaebb 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -2015,7 +2015,7 @@ MoveLayerToGroup (int layer, int group)
 {
   int prev, i, j;
 
-  if (layer < 0 || layer > max_copper_layer + 1)
+  if (layer < 0 || layer >= max_copper_layer + 2)
     return -1;
   prev = GetLayerGroupNumberByNumber (layer);
   if ((layer == bottom_silk_layer
